legend-engine-language-pure-grammar/src/test/java/org/finos/legend/engine/language/pure/grammar/test/roundtrip/TestDomainGrammarRoundtrip.java

```diff
@@ -675,4 +675,40 @@ public void testMetaFunctionExecutionWithFullPath()
                         "}\n";
         test(code);
     }
+
+    @Test
+    public void testLambdaWithBiTemporalClass()
+    {
+        test("###Pure\n" +
+                "Class <<temporal.bitemporal>> main::Person\n" +
+                "{\n" +
+                "  name: String[1];\n" +
+                "  firm: main::Firm[1];\n" +
+                "}\n" +
+                "\n" +
+                "Class <<temporal.bitemporal>> main::Firm\n" +
+                "{\n" +
+                "  name: String[1];\n" +
+                "}\n" +
+                "\n" +
+                "function main::walkTree(): main::Person[*]\n" +
+                "{\n" +
+                "   main::Person.all(%2020-12-12, %2020-12-12)\n" +
+                "}\n" +
+                "\n" +
+                "function main::walkTree1(): main::Person[*]\n" +
+                "{\n" +
+                "   main::Person.all(%latest, %latest)\n" +
+                "}\n" +
+                "\n" +
+                "function main::walkTree2(): main::Person[*]\n" +
+                "{\n" +
+                "   main::Person.all(%latest, %2020-12-12)\n" +
+                "}\n" +
+                "\n" +
+                "function main::walkTree3(): main::Firm[*]\n" +
+                "{\n" +
+                "   main::Person.all(%2020-12-12, %2020-12-12).firm(%2020-12-12, %2020-12-12)\n" +
```

Comment:
Shouldn't your test have different dates for processing and business? To ensure they are not mixed?

Reply:
Thanks update the tests